libsodium-sys/build.rs

```diff
@@ -19,12 +22,16 @@ fn main() {
 
         println!("cargo:rustc-link-lib={0}=sodium", mode);
     } else {
-        pkg_config::find_library("libsodium").unwrap();
+        if !pkg_config::probe_library("libsodium").is_ok() && !try_vcpkg() {
+            panic!("Could not find libsodium on this system!")
+        }
     }
 
     let include_dir = match env::var("SODIUM_INC_DIR") {
         Ok(dir) => dir,
-        Err(_) => pkg_config::get_variable("libsodium", "includedir").unwrap()
+        Err(_) => pkg_config::get_variable("libsodium", "includedir").unwrap_or_else(|_| {
+            get_vcpkg_include_path().unwrap()
```

Comment:
For all non msvc targets, running `get_vcpkg_include_path()` is pointless, whereas for Windows users that has specified `SODIUM_INC_DIR` will then incur the overhead of calling out to vcpkg. I chose `.unwrap_or_else()` to avoid doing that unless needed, but obviously performance isn't a huge concern in `build.rs`. If you'd prefer readability we can use `.unwrap_or()` instead.

Reply:
i just feel like we are using `unwrap_or_else` to do something that `unwrap_or` does.